File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingQuickStartButtons.java
package kore.botssdk.models;

import java.io.Serializable;
import java.util.ArrayList;

import kore.botssdk.utils.StringUtils;

public class BrandingQuickStartButtons implements Serializable {
    private Boolean show;
    private String style;
    private ArrayList<BrandingQuickStartButtonButtonsModel> buttons;
    private String input;
    private BrandingQuickStartButtonActionModel action;

    public BrandingQuickStartButtons updateWith(BrandingQuickStartButtons configModel) {
        show = configModel.show != null ? configModel.show : show;
        style = !StringUtils.isNullOrEmpty(configModel.style) ? configModel.style : style;
        input = !StringUtils.isNullOrEmpty(configModel.input) ? configModel.input : input;
        action = action != null && configModel.action != null ? action.updateWith(configModel.action) : action;
        return this;
    }

    public void setAction(BrandingQuickStartButtonActionModel action) {
        this.action = action;
    }

    public void setInput(String input) {
        this.input = input;
    }

    public BrandingQuickStartButtonActionModel getAction() {
        return action;
    }

    public String getInput() {
        return input;
    }

    public boolean isShow() {
        return show;
    }

    public String getStyle() {
        return style;
    }

    public void setShow(boolean show) {
        this.show = show;
    }

    public void setStyle(String style) {
        this.style = style;
    }

    public void setButtons(ArrayList<BrandingQuickStartButtonButtonsModel> buttons) {
        this.buttons = buttons;
    }

    public ArrayList<BrandingQuickStartButtonButtonsModel> getButtons() {
        return buttons;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingQuickStartButtons.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingQuickStartButtons(
    @SerializedName("show")
    val show: Boolean?,
    @SerializedName("style")
    val style: String = "",
    @SerializedName("buttons")
    val buttons: ArrayList<BrandingQuickStartButtonButtonsModel> = ArrayList(),
    @SerializedName("input")
    val input: String = "",
    @SerializedName("action")
    val action: BrandingQuickStartButtonActionModel?
)

fun BrandingQuickStartButtons.updateWith(configModel: BrandingQuickStartButtons?): BrandingQuickStartButtons {
    return this.copy(
        show = configModel?.show ?: show,
        style = style.ifEmpty { style },
        input = input.ifEmpty { input },
        action = action?.updateWith(configModel?.action) ?: action
    )
}